src/arithmetic.rs

```diff
@@ -70,6 +70,10 @@ where
 }
 
 /// This is a 128-bit verifier challenge.
+///
+/// The verifier samples its challenge here as u^2, i.e. the square of the
```

Comment:
These details are specific to the `Challenge` as it's used in the commitment scheme, but there are other challenges in PLONK that use this structure too. This comment can be moved in the commitment scheme implementation file.